zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt

```diff
@@ -39,14 +40,39 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
     version = BuildConfig.KOTLIN_PLUGIN_VERSION,
   )
 
+  override fun apply(target: Project) {
+    super.apply(target)
+
+    // If the build has compileDevelopmentExecutableKotlinJs, create developmentExecutableZipline.
+    // Symmetrically for production.
+    target.tasks.whenTaskAdded { task ->
+      if (task !is KotlinJsCompile) {
+        return@whenTaskAdded
+      }
+      val compileType = when (task.name) {
+        "compileDevelopmentExecutableKotlinJs" -> "developmentExecutable"
+        "compileProductionExecutableKotlinJs" -> "productionExecutable"
+        else -> return@whenTaskAdded
+      }
+      target.tasks.create("${compileType}Zipline", ZiplineCompileTask::class.java) { createdTask ->
+        createdTask.inputDir.set(
+          target.file("${target.buildDir}/compileSync/main/$compileType/kotlin")
```

Comment:
If this is a task output we ideally would get this as a property from that task so that it's both dynamic and implicitly creates a task dependency. Kotlin source are impossible to navigate on GitHub so I couldn't quickly find where this comes from on a target (presumbly it's an output of the main compilation) but we can track it down in the IDE.

Reply:
I spend 30 minutes trying to convert `KotlinJsCompile` outputs to be my Zipline compile inputs, both instances of `DirectoryProperty`. Eventually I gave up.

Reply:
OMG I had no idea. I end up not needing that cause I’m starting with configuration and not tasks, though I might be able to make that work for more robustness.